tests/test_rst_parse.py

```diff
@@ -0,0 +1,124 @@
+            cls.rst_parser = Parser(max_acts=1, max_states=1, n_best=1)
+            cls.rst_parser.load_model(str(cls.currdir / "models"))
+
+    def test_segment_and_parse(self):
+        """Test segmentation + rst parsing pipeline."""
+        if (self.parser is None or
+                self.segmenter is None or
+                self.rst_parser is None):
+            raise SkipTest("one or more models could not be found")
+
+        # create the document dictionary and run the segmenter+parser
+        with open(self.document_file, 'r') as docfh:
+            doc_dict = {"raw_text": docfh.read(), "doc_id": "test"}
+            edu_tokens, rst_trees = segment_and_parse(doc_dict,
+                                                      self.parser,
+                                                      self.segmenter,
+                                                      self.rst_parser)
+
+        # check that the document dictionary now has other fields populated
+        ok_("tokens" in doc_dict)
+        eq_(len(doc_dict["tokens"]), 14)
+        ok_("token_tree_positions" in doc_dict)
+        eq_(len(doc_dict["token_tree_positions"]), 14)
+        ok_("pos_tags" in doc_dict)
+        eq_(len(doc_dict["pos_tags"]), 14)
+        ok_("syntax_trees" in doc_dict)
+        eq_(len(doc_dict["syntax_trees"]), 14)
+        ok_("edu_start_indices" in doc_dict)
+        eq_(len(doc_dict["edu_start_indices"]), 32)
+        ok_("edu_starts_paragraph" in doc_dict)
+        eq_(len(doc_dict["edu_starts_paragraph"]), 32)
+
+        # make sure there are the expected number of EDUs
+        # and check the first and last EDUs as well
+        eq_(len(edu_tokens), 32)
+        eq_(edu_tokens[0], ['Rhetorical', 'structure', 'theory'])
+        eq_(edu_tokens[-1], ['where', 'a', 'satellites', 'have', 'been',
+                             'deleted', 'can', 'be', 'understood', 'to',
+                             'a', 'certain', 'extent', '.'])
+
+        # check that we got one tree back
+        rst_tree_list = list(rst_trees)
+        eq_(len(rst_tree_list), 1)
+        tree = rst_tree_list[0]['tree']
+        isinstance(tree, ParentedTree)
+        eq_(tree[0].label(), 'nucleus:span')
+
+    def test_segment_and_parse_partial(self):
+        """Test pipeline with partially populated document dictionary."""
+        if self.segmenter is None or self.rst_parser is None:
+            raise SkipTest("one or more models could not be found")
+
+        with open(self.partial_doc_dict_file, "r") as docdictfh:
+            doc_dict = json.load(docdictfh)
+
+        edu_tokens, rst_trees = segment_and_parse(doc_dict,
+                                                  self.parser,
+                                                  self.segmenter,
+                                                  self.rst_parser)
+
+        # check that the document dictionary now has other fields populated
+        ok_("edu_start_indices" in doc_dict)
```

Comment:
can we check these fields are not present in `doc_dict` before we call `segment_and_parse`? that catches unknowingly updating the test data later on.

Reply:
Sorry, I am not sure what you mean. Can you give me an example?

Reply:
Here's what @bndgyawali meant: what if someone adds that field to partial_doc_dict.json ? So we should also check that it wasn't there before we call `segment_and_parse()`?

Reply:
right.